main.go

```diff
@@ -73,26 +61,27 @@ func main() {
 		},
 	}
 
-	// If the user has changed the port via environment variable, respect this.
-	var portToUse string
-	if *port == 8080 {
-		portToUse = os.Getenv("PROXY_APP_PORT")
-		if portToUse == "" {
-			portToUse = fmt.Sprintf("%d", *port)
-		}
-	} else {
-		portToUse = fmt.Sprintf("%d", *port)
-	}
-
 	// Debug messaging.
-	fmt.Printf("Proxy configured to use port %s\n", portToUse)
+	fmt.Printf("Proxy configured to use port %s\n", PROXY_APP_PORT)
 	for _, target := range targets {
-		fmt.Printf("Starting proxy on http://localhost:%s for endpoint %s\n", portToUse, target.URL)
+		fmt.Printf("Starting proxy on http://localhost:%s for endpoint %s\n", PROXY_APP_PORT, target.URL)
+	}
+
+	// Handle Authentication
+	if PROXY_APP_USERNAME != "" && PROXY_APP_PASSWORD != "" {
+		e.Use(middleware.BasicAuth(func(username, password string, c echo.Context) (bool, error) {
+			// Be careful to use constant time comparison to prevent timing attacks
+			if subtle.ConstantTimeCompare([]byte(username), []byte(PROXY_APP_USERNAME)) == 1 &&
+				subtle.ConstantTimeCompare([]byte(password), []byte(PROXY_APP_PASSWORD)) == 1 {
+				return true, nil
+			}
+			return false, nil
+		}))
 	}
 
 	// Start serving the proxy as configured.
 	e.Use(middleware.Proxy(middleware.NewRoundRobinBalancer(targets)))
-	localAddress := fmt.Sprintf(":%s", portToUse)
+	localAddress := fmt.Sprintf(":%s", PROXY_APP_PORT)
 	e.Logger.Fatal(e.Start(localAddress))
```

Comment:
I wonder if `e.Start()` validates the `localAddress`. That way you could just use `os.Get("PROXY_APP_ADDRESS")` here and have a default in the Docker image.

Reply:
Reworked how the port is used - removed localAddress.
I have checked and no, no validation occurs.